networking/private-cloud-function-from-onprem/main.tf

```diff
@@ -14,80 +14,60 @@
  * limitations under the License.
  */
 
-###############################################################################
-#                                   locals                                    #
-###############################################################################
 locals {
-  prefix = var.prefix != null ? "${var.prefix}-" : ""
+  psc_name = replace(var.name, "-", "")
 }
 
-###############################################################################
-#                                  projects                                   #
-###############################################################################
-
-module "project-onprem" {
+module "project" {
   source          = "../../modules/project"
-  billing_account = var.billing_account_id
-  name            = var.projects_id.onprem
-  parent          = var.root_node
-  project_create  = var.create_projects
-  prefix          = var.prefix
+  name            = var.project_id
+  project_create  = var.project_create == null ? false : true
+  billing_account = try(var.project_create.billing_account_id, null)
+  parent          = try(var.project_create.parent, null)
+  service_config = {
+    disable_dependent_services = false
+    disable_on_destroy         = false
+  }
   services = [
+    "cloudfunctions.googleapis.com",
+    "cloudbuild.googleapis.com",
     "compute.googleapis.com",
     "dns.googleapis.com"
   ]
 }
 
-
-module "project-hub" {
-  source          = "../../modules/project"
-  billing_account = var.billing_account_id
-  name            = var.projects_id.function
-  parent          = var.root_node
-  project_create  = var.create_projects
-  prefix          = var.prefix
-  services = [
-    "compute.googleapis.com",
-    "cloudfunctions.googleapis.com",
-    "cloudbuild.googleapis.com"
-  ]
-}
-
 ###############################################################################
 #                                  VPCs                                       #
 ###############################################################################
 
 module "vpc-onprem" {
   source     = "../../modules/net-vpc"
-  project_id = module.project-onprem.project_id
-  name       = "${local.prefix}onprem"
+  project_id = module.project.project_id
+  name       = "${var.name}-onprem"
   subnets = [
     {
       ip_cidr_range      = var.ip_ranges.onprem
-      name               = "${local.prefix}onprem"
+      name               = "${var.name}-onprem"
       region             = var.region
       secondary_ip_range = {}
     }
   ]
```

Comment:
I'd remove Private Google Access from this network, otherwise the function would be reachable even without the VPN